src/mender-update/daemon/states.cpp

```diff
@@ -173,8 +238,10 @@ void PollForDeploymentState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &
 	auto err = ctx.deployment_client->CheckNewDeployments(
 		ctx.mender_context,
 		ctx.http_client,
-		[&ctx, &poster](mender::update::deployments::CheckUpdatesAPIResponse response) {
+		[this, &ctx, &poster](mender::update::deployments::CheckUpdatesAPIResponse response) {
 			if (!response) {
+				// Replace the update poll timer with a backoff
```

Comment:
One important thing,
Here and above (commenting here because it is easier to see on the following lines),

When the error is `auth::UnauthorizedError`, we shall not trigger the retry mechanism! That must be the exception.